client/src/components/Today/timeline/CreateInlineActivity.tsx

```diff
@@ -0,0 +1,191 @@
+import { Checkbox } from "@/components/utility/Checkbox/Checkbox";
+import { createDate } from "@/lib/datetime/make-date";
+import { useBreakpoints } from "@/lib/hooks/breakpoints";
+import type { MainTheme } from "@/lib/style/theme";
+import Buttons from "@/lib/theme/components/buttons";
+import Containers from "@/lib/theme/components/container.style";
+import { font } from "@/lib/theme/font";
+import { spacingValue } from "@/lib/theme/snippets/spacing";
+import S from "./style/CreateInlineActivity.style";
+
+// TODO: min, max, presets (also for the other time picker)
+const timePickerProps = (isMobileWidth: boolean): TimePickerProps => ({
+	withDropdown: true,
+	minutesStep: 5,
+	popoverProps: {
+		withinPortal: false,
+		position: "bottom",
+	},
+	w: "9ch",
+	flex: isMobileWidth ? 1 : 0,
+	style: {
+		touchAction: "none",
+		userSelect: "none",
+	},
+});
+
+// TODO: go over styling. What goes in styled components, and what needs to
+// remain inline?
+export default function CreateInlineActivity({
+	date,
+	timelineRowIndex,
+	ref,
+	active,
+}: {
+	date: Datelike;
+	timelineRowIndex: number;
+	ref: RefObject<HTMLDivElement | null>;
+	active: null | number;
+}) {
+	const theme = useTheme() as MainTheme;
+	const {
+		activity,
+		isValidActivity,
+		setActivity,
+		handleInputChange,
+		handleSubmit,
+	} = useActivityForm({
+		inline: true,
+		date,
+		timelineRowIndex,
+	});
+	const { isMobileWidth } = useBreakpoints();
+
+	/** the regular ActivityForm uses datetime pickers, hence we need a new
+	 * handler for the time-only inputs in this component. */
+	function handleTimeChange(value: string, type: "start" | "end") {
+		const field =
+			type === "start"
+				? "started_at"
+				: ("ended_at" as const satisfies keyof Activity);
+
+		const [hour, minute] = value.split(":");
+
+		setActivity(
+			produce((draft) => {
+				draft[field] = createDate(date).hour(+hour).minute(+minute);
+			})
+		);
+	}
+
+	return (
+		active === timelineRowIndex &&
+		!!activity.started_at &&
+		!!activity.ended_at && (
+			<S.Pop
+				ref={ref}
+				layout
+				layoutId="timeline-new-activity"
+				key={`timeline-popover-${timelineRowIndex}`}
+				style={
+					{
+						willChange: "auto",
+						outline: `2px solid ${theme.colors.background.main[3]}`,
+						zIndex: 300,
+						backgroundColor: theme.colors.background.main[0],
+						fontSize: font.size["0.85"],
+						borderRadius: 3,
+						display: "flex",
+						flexDirection: "row",
+					} as CSSProperties
+				}
+				variants={timelinePopoverMotionVariants(theme)}
+				initial="initial"
+				// this thing is only visible when active === index, so we can
+				// always assume we're "animating" the "active" variant.
+				animate="active"
+				transition={{
+					duration: 0.035,
+					ease: "easeInOut",
+				}}
+			>
+				<Containers.Row style={{ alignItems: "stretch" }} gap="smaller">
+					<motion.div
```

Comment:
I don't like the idea of making a styled component for this, but we could put the style object in a variable.